packages/SwingSet/src/liveslots/collectionManager.js

```diff
@@ -509,10 +515,62 @@ export function makeCollectionManager(
       return iter();
     }
 
+    /**
+     * Clear the entire contents of a collection non-selectively.  Since we are
+     * being unconditional, we don't need to inspect any of the keys to decide
+     * what to do and therefor can avoid deserializing the keys.  In particular,
+     * this avoids swapping in any virtual objects that were used as keys, which
+     * can needlessly thrash the virtual object cache when an entire collection
+     * is being deleted.
+     *
+     * @returns {boolean} true if this operation introduces a potential
+     *   opportunity to do further GC.
+     */
+    function clearInternalFull() {
+      let doMoreGC = false;
+      const [coverStart, coverEnd] = getRankCover(M.any(), encodeKey);
+      let priorDBKey = '';
+      const start = prefix(coverStart);
+      const end = prefix(coverEnd);
+      while (priorDBKey !== undefined) {
+        const [dbKey, dbValue] = syscall.vatstoreGetAfter(
+          priorDBKey,
+          start,
+          end,
+        );
+        if (!dbKey) {
+          break;
+        }
+        if (dbKey < end) {
+          priorDBKey = dbKey;
+          const value = JSON.parse(dbValue);
+          doMoreGC =
+            value.slots.map(vrm.removeReachableVref).some(b => b) || doMoreGC;
+          syscall.vatstoreDelete(dbKey);
+          if (dbKey[0] === 'r') {
```

Comment:
This is coupling to the behavior of.. `packages/store/src/patterns/encodePassable.js > makeEncodePassable()`? I'm a bit nervous about bitrot if `store` decided to change their encoding scheme.

I obviously don't want to call `decodeRemotable()` on `dbKey`, that would defeat the purpose of this fix. 

One option would be to create a separate instance of `makeDecodePassable()` whose only job is to report "reference or not", probably by counting whether the `decodeRemotable` passed in was called or not:
```js
function dbKeyIsRemotable(dbKey) {
  let isRemotable = false;
  function decodeRemotable(rem) { isRemotable = true; return undefined; }
  const decoder = makeDecodePassable({decodeRemotable});
  decoder(dbKey);
  return isRemotable;
}
```

Another option is to add a unit test which explicitly checks the output of `makeEncodePassable` to see that the encoded key starts with `'r'`, and to add cross-pointing comments between the test and this line (so `store` ever changes its behavior, our test will fail, and when someone goes to update the test, they'll following the pointer here and update the `=== 'r'` comparison at the same time).

A third option (the most efficient and maintainable) would be to change `store` to export an additional function like:
```js
export const encodedIsRemotable = (encoded) => encoded[0] === 'r';
harden(encodedIsRemotable);
```

plus a unit test in `store`. Not sure what @erights would think about that.

I'm undecided as to which I like better, but I know I want at least one of them.

Reply:
My first reaction was that this ship had already sailed, as we have various other code that has baked in understanding of the encoding, but closer inspection suggests that's not so (there _is_ some code like that but it's already inside the `store` package where it belongs).  Of the three options you propose, I think the first is gross and the second is icky, so I lean towards the third.  There's already all kinds of other whacky stuff exported from `store` so I don't expect @erights to object too loudly.  In the interest of moving things forward I'm inclined to follow the "ask for forgiveness rather than permission" strategy.